mrp_subcontracting/models/mrp_production.py

```diff
@@ -8,36 +8,40 @@
 class MrpProduction(models.Model):
     _inherit = 'mrp.production'
 
-    @api.one
-    def _created_purchases(self):
-        cond = [('mrp_production', '=', self.id)]
-        self.created_purchases = len(self.env['purchase.order'].search(cond))
-
-    @api.one
-    def _created_outpickings(self):
-        picking_obj = self.env['stock.picking']
-        cond = [('mrp_production', '=', self.id)]
-        self.created_outpickings = len(
-            picking_obj.search(cond).filtered(
-                lambda x: x.picking_type_id.code == 'outgoing'))
-
-    @api.one
-    def _created_inpickings(self):
-        picking_obj = self.env['stock.picking']
-        cond = [('mrp_production', '=', self.id)]
-        self.created_outpickings = len(
-            picking_obj.search(cond).filtered(
-                lambda x: x.picking_type_id.code == 'incoming'))
-
     created_purchases = fields.Integer(
         string='Created Purchases', readonly=True,
-        compute='_created_purchases', track_visibility='always')
+        compute='_compute_created_purchases', track_visibility='always')
     created_outpickings = fields.Integer(
         string='Created Out Pickings', readonly=True,
-        compute='_created_outpickings', track_visibility='always')
+        compute='_compute_created_outpickings', track_visibility='always')
     created_inpickings = fields.Integer(
         string='Created In Pickings', readonly=True,
-        compute='_created_inpickings', track_visibility='always')
+        compute='_compute_created_inpickings', track_visibility='always')
+
+    @api.multi
+    def _compute_created_purchases(self):
+        PurchaseOrder = self.env['purchase.order']
```

Comment:
This will be marked by the linter as invalid (although semantically is good). Please change to `purchase_order_model` or `purchase_order_obj` or `po_obj`.